### PR TITLE
Added dynamic Piece Hierarchy

### PR DESCRIPTION
**Pull request by PicassoCT**
this allows for dynamic rearranging of pieces via the LuaSyncedCtrl Call SetUnitPieceHierarchy().

The function  SetUnitPieceHierarchy() takes the Unitid, a HeadPiece, and a Table with Key/Value Pairs, where the  keys are the Parents and the Values (Tables) are the children.

It compiled, but i need a windows build for further debugging.

### CODE REVIEW

File: rts/Lua/LuaSyncedCtrl.h [hunk 1]
```diff
@@ -174,5 +176,21 @@ class LuaSyncedCtrl
 		static int SetRadarErrorParams(lua_State* L);
 };
 
+//Stores Pieces Hierarchy Information - HigherPiece is the Parent, SubPieces the direct Children
+struct PieceHierarchy
+{
+	float HigherPiece;
+	std::vector<float> * SubPieces;
```
Comment (abma) [inline rts/Lua/LuaSyncedCtrl.h]: i didn't really investigate the code, but at a quick look:

1. why float? aren't piece numbers int/unsigned int/size_t?
2. why a pointer?

should do the same:
std::vector<size_t> SubPieces;

**Comment by abma** [comment]
hmm, why did you delete your pull request? are you working on an updated version?

**Comment by YangHaeKyoung** [comment]
NAVER - http://www.naver.com/
--------------------------------------------
양혜경(hk.yang@navercorp.com) 님은 현재 부재중입니다.</br>
보내신 메일 <Re: [spring] Added dynamic Piece Hierarchy (#214)> 은 저장되어 있으므로 다시 보내실 필요는 없습니다.</br>
양혜경(hk.yang@navercorp.com) 님이 남기신 메시지 입니다.
--------------------------------------------
[Absence period : 2015-08-10 00:00 ~ 2015-08-14 23:59 GMT +09:00]<br>휴가입니다. 급한 문의는 핸드폰으로 연락 주세요. 감사합니다.
--------------------------------------------

**Comment by abma** [comment]
@YangHaeKyoung: when you come back, it would be nice when you disable your autoreply :D

**Comment by PicassoCT** [comment]
Yes.. i replaced this with a  (by now working and tested) Spring.SetUnitPieceParent(unitID, piece,pieceparent) 
Im still trying to stage a decent commit - but develop is moving fast this day.

**Pull request closed without merging**